File: C/ComplexNumber.c
#include<stdio.h>
struct complex
{
    float real;
    float imag;
};

struct complex input(float x, float y);
void display(struct complex p);

int main()
{
    struct complex c = {10.3, 20.6};                    //structure initialization
    printf("Displaying for the first time:\n");
    display(c);
    c = input(12.5F, 15.5F);
    printf("Displaying after input():\n");
    display(c);
}

struct complex input(float x, float y)
{
    struct complex r;            
    r.real = x;
    r.imag = y;
    return r;
}

void display(struct complex p)
{     
    printf("real: %.1f\n", p.real);
    printf("imag: %.1f\n", p.imag);
}

/*
    Add two complex numbers
*/

File: Java/ComplexNumber.java
class Complex {
    private float real;
    private float imag;

    public Complex() {
        real = 0.0F;
        imag = 0.0F;
    }

    public Complex(float x, float y) {
        this.real = x;
        imag = y;
    }

    public void input(float x, float y) {
        real = x;
        imag = y;
    }

    public void display()
    {
        System.out.println("real: " + this.real);
        System.out.println("imag: " + this.imag);
    }

    Complex add(Complex p) {
        Complex t = new Complex();
        t.real = real + p.real;
        t.imag = imag + p.imag;
        return t;
    }
}

class ComplexNumber {
    public static void main(String[] args) {
        Complex c1 = new Complex(10.3F, 20.6F);
        Complex c2 = new Complex(25.6F, 99.7F);
        Complex c3;     //Object Reference Declaration
        c3 = c1.add(c2);    //C3 initialized here
        //c3 = c1 + c2;     Not supported in java
        c1.display();
        c2.display();
        c3.display();
    }
}